src/webob/acceptparse.py

```diff
@@ -1014,619 +822,97 @@ def acceptable_offers(self, offers):
-        of the offer (if the server quality value is not supplied, it is 1).
+        :param default_match:
+            (optional, any type) the value to be returned if there is no match
 
-        The offer in the match with the highest quality value is the best
-        match. If there is more than one match with the highest qvalue, the
-        match where the media range has a lower number of '*'s is the best
-        match. If the two have the same number of '*'s, the one that shows up
-        first in `offers` is the best match.
+        :return:
+            (``str``, or the type of ``default_match``)
 
-        :param offers: (iterable)
+            | The offer that is the best match based on q-value. If there is no
+              match, the value of ``default_match`` is returned. Where two
+              offers match the same qvalue, they are returned in the same order
+              as their order in ``offers``.
 
-                       | Each item in the iterable may be a ``str`` media type,
-                         or a (media type, server quality value) ``tuple`` or
-                         ``list``. (The two may be mixed in the iterable.)
+        .. versionchanged:: 2.0
 
-        :param default_match: (optional, any type) the value to be returned if
-                              there is no match
+            - Offers containing wildcards are no longer allowed.
+            - A tuple can no longer be an offer containing server-side quality
+              values.
+            - An offer will only match a wildcard clause in the header, such
+              as ``*/*`` or ``text/*`` if it does not match something more
+              specific.
+        """
+        matches = self.acceptable_offers(offers)
+        if matches:
+            return matches[0][0]
+        return default_match
 
-        :return: (``str``, or the type of `default_match`)
+    def quality(self, offer):
+        """
+        Return quality value of given ``offer``, or ``None`` if there is no match.
 
-                 | The offer that is the best match. If there is no match, the
-                   value of `default_match` is returned.
+        This is a thin wrapper around :meth:`.acceptable_offers` that matches
+        a specific ``offer``.
 
-        This uses the old criterion of a match in
-        :meth:`AcceptValidHeader._old_match`, which is not as specified in
-        :rfc:`RFC 7231, section 5.3.2 <7231#section-5.3.2>`. It does not
-        correctly take into account media type parameters:
+        :param offer: (``str``) media type offer
+        :return: (``float`` or ``None``)
 
-            >>> instance = AcceptValidHeader('text/html')
-            >>> instance.best_match(offers=['text/html;p=1']) is None
-            True
+                 | The highest quality value from the media range(s) that match
+                   the `offer`, or ``None`` if there is no match.
```

Comment:
Would be helpful to copy over the documentation for the return value in `_AcceptInvalidOrNoHeader.quality`, to explain what happens when the header is invalid or missing?